systemd/crc-dnsmasq.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -x
+
+# exit early if user-mode networking
+ping -c1 gateway > /dev/null 2>&1
+[ $? -eq 0 ] && exit 0
+
+hostName=$(hostname)
+ip=$(ip -4 addr show br-ex | grep -oP '(?<=inet\s)192+(\.\d+){3}')
+iip=$(hostname -i)
+
+cat << EOF > /etc/dnsmasq.d/crc-dnsmasq.conf
+listen-address=$ip
+expand-hosts
+log-queries
+local=/crc.testing/
+domain=crc.testing
+address=/apps-crc.testing/$ip
+address=/api.crc.testing/$ip
+address=/api-int.crc.testing/$ip
+address=/$hostName.crc.testing/$iip
```

Comment:
why we need this one for node name resolve (in OCP context) Also I think `ip` should be node name dummy IP which is 192.168.126.11?

Reply:
yeah probably not needed now, i think this was needed before moving to SNO and in the [crc code-base](https://github.com/crc-org/crc/blob/f4ccc5891d66148cf5c9c2361b2784b7b3f10aa4/pkg/crc/services/dns/template.go#L19) we still have it so added here as well

Reply:
did we test it with dummy IP `192.168.126.11` ?

Reply:
you mean using `address=/$hostName.crc.testing/192.168.126.11`? instead of use the result from `hostname -i`?

Reply:
yes right, because this is used by internal service resolution so it might work.